src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,29 @@
 import React from 'react';
+import { RootState } from '../../app/store';
+import { actions as filterActions } from '../../features/filter';
+import { useDispatch, useSelector } from 'react-redux';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useDispatch();
+
+  const filters = useSelector((state: RootState) => state.filter);
+
   return (
     <form
+      name="searchQuery"
       className="field has-addons"
-      onSubmit={event => event.preventDefault()}
+      onSubmit={event => {
+        event.preventDefault();
+      }}
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
+          <select
+            data-cy="statusSelect"
+            onChange={event => {
+              dispatch(filterActions.addStatus(event.target.value));
```

Comment:
Ensure that `filterActions.addStatus` is the correct action for updating the status filter. If the action is named differently in your Redux slice, update it accordingly.